pkg/controller/publishingstrategy/publishingstrategy_controller.go

```diff
@@ -0,0 +1,217 @@
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// Reconcile will handle
+	// 1. Set the cluster API to Internal
+	// 2. Set the cluster API to External (Internet-facing)
+	// 3. Set the default ingress (application) to Internal
+	// 4. Set the default ingress (application) to External (Internet-facing)
+
+	// get region
+	region, err := utils.GetClusterRegion(r.client)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	// Secret should exist in the same namespace Account CR's are created
+	awsClient, err := awsclient.GetAWSClient(r.client, awsclient.NewAwsClientInput{
+		SecretName: config.AWSSecretName,
+		NameSpace: 	config.OperatorNamespace,
+		AwsRegion: 	region
+	})
+	if err != nil {
+		reqLogger.Error(err, "Failed to get AWS client")
+		return reconcile.Result{}, err
+	}
+
+	// To make API server non-public
+	// 1. Delete ext NLB for port 6443/TCP
+	// 2. Update api.<cluster-domain> DNS record to point to internal NLB
+	if instance.Spec.DefaultAPIServerIngress.Listening == cloudingressv1alpha1.Internal {
+		// get a list of load balancers in use
+		descLBInput := elbv2.DescribeLoadBalancersInput{}
+		descLBOutput, err := awsClient.DescribeLoadBalancersV2(&descLBInput)
+		if err != nil {
+			log.Error(err, "Error getting list of NLBs")
+			return reconcile.Result{}, err
+		}
+		
+		// get the internet-facing NLB by scheme and delete
+		var internalNLBDNS *string
+		var internalCanonicalHostedZoneID *string
+		for _, loadBalancers := range descLBOutput.LoadBalancers{
+			external := "internet-facing"
+			scheme := *loadBalancers.Scheme
+			if scheme == external {
+				//need to get Amazon Resource Name (ARN) of the load balancer to pass in input
+				extARN := loadBalancers.LoadBalancerArn
+				deleteLBInput := elbv2.DeleteLoadBalancerInput{extARN}
+				deleteLBOutput, err := awsClient.DeleteLoadBalancerV2(&deleteLBInput)
+				if err != nil {
+					log.Error(err, "Error deleting internet-facing load balancer")
+					return reconcile.Result{}, err
+				}
+			}
+			if scheme == "internal" {
+				internalCanonicalHostedZoneID = loadBalancers.CanonicalHostedZoneId
+				internalNLBDNS = loadBalancers.DNSName
+			}
+		}
+
+		// change Alias of external LB in public hosted zone to internal LB
+
+		dnsSpec := configv1.DNSSpec{}
+		domainName := dnsSpec.BaseDomain // in form of ```samn-test.j5u3.s1.devshift.org```
+
+		// get index of first period
+		perIndex := strings.Index(domainName, ".")		
+		// get pubDomainName in form of ```j5u3.s1.devshift.org```
+		pubDomainName := domainName[perIndex+1:len(domainName)]
```

Comment:
I think it's okay to combine  these? `pubDomainName := domainName[strings.Index(domainName, ".")+1:len(domainName)]` doesn't seem too unwieldy? What do you think?  If we have a good comment explaining why it's important to do this operation on the cluster domain name. (eg, "in order to update DNS (this would be one of the "list of steps" earlier), we need the route53 zone name, which happens to be the cluster domain minus the name of the cluster" or something like that).

Reply:
I agree. As long as the comment is good I don't see an issue with a one-liner.